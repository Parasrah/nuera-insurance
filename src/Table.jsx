import React from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, css } from 'aphrodite'

import { Row } from '@/Row'

function Table ({ addRow, total, map, mapRow }) {
  return (
    <div className={css(styles.mobileSize, styles.desktopSize, styles.blue)}>
      {map(r => <Row {...r} />)}
    </div>
  )
}

const breakpoint = 800

const styles = StyleSheet.create({
  desktopSize: {
    [`@media (min-width: ${breakpoint + 1}px)`]: {
      width: '50%',
      height: '70%',
      marginBottom: '5%'
    }
  },
  mobileSize: {
    [`@media (max-width: ${breakpoint}px)`]: {
      height: '100%',
      width: '100%',
      margin: 0
    }
  },
  blue: {
    backgroundColor: 'blue'
  }
})

Table.propTypes = {
  addRow: PropTypes.func.isRequired,
  total: PropTypes.func.isRequired,
  map: PropTypes.func.isRequired,
  mapRow: PropTypes.func.isRequired
}

export { Table }
