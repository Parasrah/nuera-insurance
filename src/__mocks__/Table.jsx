import React from 'react'

function Table () {
  return <div className='table' />
}

export { Table }
