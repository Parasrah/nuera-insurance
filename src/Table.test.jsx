/* globals describe, test, expect, jest */
import React from 'react'
import renderer from 'react-test-renderer'

import { Table } from '@/Table'
jest.mock('@/Row')

function getHookMocks () {
  return {
    addRow: jest.fn(),
    getTotal: jest.fn().mockReturnValue(687.79),
    fromCategories: jest.fn().mockReturnValue([]),
    categoryTotal: jest.fn().mockReturnValue(66.99),
    deleteRow: jest.fn(),
    getCategories: jest.fn().mockReturnValue(['Electronics'])
  }
}

describe('#render', () => {
  test('renders correctly', () => {
    const hookMocks = getHookMocks()
    const tree = renderer
      .create(<Table {...hookMocks} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
