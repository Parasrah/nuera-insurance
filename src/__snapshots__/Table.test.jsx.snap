// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render renders correctly 1`] = `
<div
  className="mobileSize desktopSize table"
>
  <div
    className="categories"
  >
    <div
      className="category"
    >
      <div
        className="rowContainer row false"
      >
        <span
          className="name grow"
        >
          Total
        </span>
        <span
          className="value grow"
        >
          ($
          687.79
          )
        </span>
        <div
          className="filler"
        />
        <div
          className="line"
        />
      </div>
    </div>
  </div>
  <div
    className="filler"
  />
  <div
    className="bottom"
  >
    <div
      className="add"
    >
      <input
        className="input name"
        id="name"
        name="name"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Name"
        value=""
      />
      <label
        className="label"
        htmlFor="name"
      >
        name
      </label>
      <input
        className="input "
        id="value"
        name="value"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Value ($)"
        value=""
      />
      <label
        className="label"
        htmlFor="value"
      >
        value
      </label>
      <div
        className="dropdown dropdown"
        onClick={[Function]}
      >
        <span>
          Electronics
        </span>
        <test-file-stub
          className="chevron"
          viewBox="0 0 24 24"
        />
      </div>
      <test-file-stub
        className="button"
        onClick={[Function]}
      />
    </div>
  </div>
  <div
    className="Toastify"
  />
</div>
`;
